src/webgpu/api/operation/storage_texture/read_only.spec.ts

```diff
@@ -107,6 +107,29 @@ class F extends GPUTest {
                 SetData(texelValue, outputValue, texelDataIndex, component);
                 break;
               }
+              case 'bgra8unorm': {
+                const texelValue = (4 * texelDataIndex + component + 1) % 256;
+                const texelComponentIndex = texelDataIndex * componentCount + component;
+                texelTypedDataView[texelComponentIndex] = texelValue;
+
+                const outputValue = texelValue / 255.0;
+                let outputComponent = 0;
+                switch (component) {
+                  case 0:
+                    outputComponent = 2;
+                    break;
+                  case 2:
+                    outputComponent = 0;
+                    break;
+                  case 1:
+                  case 3:
+                    outputComponent = component;
+                    break;
+                }
```

Comment:
I would suggest changing this to a look-up:
```suggestion
                const outputComponent = [2, 1, 0, 3][component];
```
Maybe with an assertion
```typescript
                  assert(component < 4);
```